smarts/core/controllers/__init__.py

```diff
@@ -136,10 +136,7 @@ def from_action_space(action_space, vehicle_pose, sim):
             ActionSpaceType.Lane,
             ActionSpaceType.LaneWithContinuousSpeed,
         ):
-            # TAI: we should probably be fetching these waypoint through the mission planner
-            target_lane_id = sim.road_network.lanepoints.closest_lanepoint(
-                vehicle_pose, filter_from_count=4
-            ).lane_id
+            target_lane_id = sim.road_map.nearest_lane(vehicle_pose.point).lane_id
             return LaneFollowingControllerState(target_lane_id)
```

Comment:
I think this is just to specify what lane it starts on.
Fwiw, I updated this since you reviewed it to add an assert and message if the `nearest_lane` call returns `None`.